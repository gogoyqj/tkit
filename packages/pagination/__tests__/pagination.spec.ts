import { parser } from './parser';

const files = 'testPaginationFail.ts testPaginationOK.ts'.split(' ');

describe('utils/pagination work ok', () => {
  const errors = parser(files, 'paginationSamples/');
  files.forEach(file =>
    it(`check sample/${file} work ok`, () => {
      const current = errors[file];
      if (file.match(/OK/)) {
        expect(current).toHaveLength(0);
      } else if (file.match(/Fail/)) {
        // expect(current.length).toBeGreaterThan(0);
      }
      expect(current).toMatchSnapshot();
    })
  );
});
