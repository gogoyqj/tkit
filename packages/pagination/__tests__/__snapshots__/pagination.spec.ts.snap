// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/pagination work ok check sample/testPaginationFail.ts work ok 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'string' is not assignable to type 'number'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'string' is not assignable to type 'boolean'.",
  },
  Object {
    "category": 1,
    "code": 2551,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Property 'userList2' does not exist on type 'Store'. Did you mean 'userList'?",
  },
  Object {
    "category": 1,
    "code": 2339,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Property 'total2' does not exist on type 'TkitPagenationStateBase<{ id: number; name: string; }, MergedParams>'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'string' is not assignable to type 'number'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'number' is not assignable to type 'string'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'true' is not assignable to type 'string | number'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'string' is not assignable to type 'number'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'number' is not assignable to type 'string'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'number' is not assignable to type 'string'.",
  },
  Object {
    "category": 1,
    "code": 2339,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Property 'userList2' does not exist on type 'Store'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'number' is not assignable to type 'string'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'string' is not assignable to type 'number'.",
  },
  Object {
    "category": 1,
    "code": 2322,
    "fileName": "paginationSamples/testPaginationFail.ts",
    "messageText": "Type 'number' is not assignable to type 'string'.",
  },
]
`;

exports[`utils/pagination work ok check sample/testPaginationOK.ts work ok 1`] = `Array []`;
